application/src/main/java/org/togetherjava/tjbot/features/help/HelpThreadCreatedListener.java

```diff
@@ -90,6 +96,28 @@ private void handleHelpThreadCreated(ThreadChannel threadChannel) {
             .queueAfter(5, TimeUnit.SECONDS);
     }
 
+    private static User getAuthorFromPostByBot(final Message postMessage) {
+
+        final List<User> users = postMessage.getMentions().getUsers();
+
+        if (users.size() != 1) {
+
+            String errorMsg = "Attempts to open a help-thread for multiple authors.";
+
+            if (users.isEmpty()) {
+                errorMsg = "Attempts to open a help-thread without any author.";
+            }
+
+            throw new IllegalStateException(errorMsg);
+        }
```

Comment:
there are many things wrong here
1. nested if statement is never reachable
2. the error message "Attempts to open...." doesn't make sense the bot is trying to close it
3. in the current state of the bot the number of pings can never exceed 1

so just check if its not 1, and throw the error, the error message should be along the lines of "Cannot find original user in help-thread: <ID>"

Reply:
ohh ig ur right, i misread the code

> I don't see a technical guarantee for that

wdym? the bot will in no  case  ping more than 1 person

Reply:
I know but what to do in case of sudden error or a misbehavior (like mentioning a wrong user or a role)? Ig throwing an exception is helpful 🤷‍♂️